CMakeTests/AddCompileDefinitions.cmake

```diff
@@ -0,0 +1,42 @@
+# Add a C or C++ compile definitions to the current scope
+#
+# add_compile_definitions(def [def ...] [DEBUG | RELEASE])
+#
+# Can optionally add the definitions to Debug or Release configurations only, use this when
+# targeting multi-configuration generators like Visual Studio or Xcode.
+# Release configurations means NOT Debug, so it will work for RelWithDebInfo or MinSizeRel too.
+# The definitions are added to the COMPILE_DEFINITIONS folder property.
+# Supports generator expressions, unlike add_definitions()
+#
+# Examples:
+#   add_compile_definitions(FOO) -> -DFOO
+#   add_compile_definitions(_DEBUG DEBUG) -> -D_DEBUG
+#   add_compile_definitions(NDEBUG RELEASE) -> -DNDEBUG
+#   add_compile_definitions($<$<COMPILE_LANGUAGE:C>:THISISONLYFORC>)
+
+function(add_compile_definitions def)
```

Comment:
This is confusingly close to a built-in CMake command for me. Could we do like `dolphin_compile_definitions(...)` or something?